File: src/CodeInTasks.Web.Models/Task/TaskViewModel.cs
namespace CodeInTasks.Web.Models.Task
{
    public class TaskViewModel
    {
        public Guid Id { get; set; }

        public string Title { get; set; }
        public string Description { get; set; }
        public TaskCategory Category { get; set; }
        public DateTime CreateDate { get; set; }

        public RunnerType Runner { get; set; }
        public string BaseRepositoryUrl { get; set; }
        public string TestRepositoryUrl { get; set; }

        public Guid CreatorId { get; set; }
    }
}

File: frontend/CodeInTasks/src/app/web-api/models/task/task-view-model.ts
import { RunnerType, TaskCategory } from "../../enums";

export class TaskViewModel {
    public id: string = null!;

    public title: string = null!;
    public description: string = null!;
    public category: TaskCategory = null!;
    public createDate: Date = null!;

    public runner: RunnerType = null!;

    public baseRepositoryUrl: string = null!;
    public testRepositoryUrl: string = null!;
    public testRepositoryAuthPassword: string = null!;

    public creatorId: string = null!;
}
